### DISCUSSION TITLE
Can not create or edit files in GitHub

### DISCUSSION THREAD

**Discussion by Flusinerd** [community/community#23221] [New to GitHub]
<p>Hello,<br>
I am unable to edit or create files in my GitHub repo from the web client.<br>
It always fails with <code>File could not be edited</code></p>
<p>I am able to push to my repository using git without issues.<br>
Also this might be related to dependabot not being able to create PRs.<br>
I have 3 outdated dependencies I want to upgrade but. One of the dependencies is failing with:<br>
<code>Invalid git modules config</code><br>
The others are failing with no specific error message. The issue is, that there is no error in the dependabot log files.</p>
<p>I have a git submodule in this repository. The file structure looks like this:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/6/0/60bebe68a8aa5915abfddc2a6013a4cc660598cd.png" data-download-href="/uploads/short-url/dNQveKmAIIL13tcz28TqXWQ7G2x.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/22977969/181075643-cdcef1ca-7279-4491-af81-092251497ac7.png" alt="image" data-base62-sha1="dNQveKmAIIL13tcz28TqXWQ7G2x" width="690" height="227" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/6/0/60bebe68a8aa5915abfddc2a6013a4cc660598cd_2_690x227.png, https://aws1.discourse-cdn.com/github/optimized/3X/6/0/60bebe68a8aa5915abfddc2a6013a4cc660598cd_2_1035x340.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/6/0/60bebe68a8aa5915abfddc2a6013a4cc660598cd_2_1380x454.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/6/0/60bebe68a8aa5915abfddc2a6013a4cc660598cd_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">1808×596 78.7 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>And my <code>.gitmodules</code> like this:</p>
<pre><code class="lang-auto">[submodule "./backend/label"]
	path = backend/label
	url = https://github.com/&lt;org&gt;/&lt;repo&gt;.git
	branch = main
</code></pre>
<p>Maybe someone can help me.</p>
<p>Sincerely,<br>
Jan</p>

**Comment by Flusinerd** [comment, marked as answer]
<p>Okay looks like I finally managed to fix it.<br>
The issue was <code>.backend/label</code> in the <code>.gitmodules</code> file. Apparently you have to omit the dot.<br>
So this works:</p>
<pre><code class="lang-auto">[submodule "backend/label"]
	path = backend/label
	url = https://github.com/&lt;org&gt;/&lt;repo&gt;.git
	branch = main
</code></pre>
<p>Sincerely,<br>
Jan</p>

**Reply by ChristmasGit** [reply]
Where do you even find this folder to change that...

**Reply by tbohermes** [reply]
Yeah, that was hyper Impressive

**Comment by Harinilalitha** [comment]
# This workflow uses actions that are not certified by GitHub.
# They are provided by a third-party and are governed by
# separate terms of service, privacy policy, and support
# documentation.

# Sample workflow for building and deploying a Jekyll site to GitHub Pages
name: Deploy Jekyll site to Pages

on:
  # Runs on pushes targeting the default branch
  push:
    branches: ["master"]

  # Allows you to run this workflow manually from the Actions tab
  workflow_dispatch:

# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
permissions:
  contents: read
  pages: write
  id-token: write

# Allow one concurrent deployment
concurrency:
  group: "pages"
  cancel-in-progress: true

jobs:
  # Build job
  build:
    runs-on: ubuntu-latest
    steps:
      - name: Checkout
        uses: actions/checkout@v3
      - name: Setup Ruby
        uses: ruby/setup-ruby@0a29871fe2b0200a17a4497bae54fe5df0d973aa # v1.115.3
        with:
          ruby-version: '2.6' # Not needed with a .ruby-version file
          bundler-cache: true # runs 'bundle install' and caches installed gems automatically
          cache-version: 1 # Increment this number if you need to re-download cached gems
      - name: Setup Pages
        id: pages
        uses: actions/configure-pages@v2
      - name: Build with Jekyll
        # Outputs to the './_site' directory by default
        run: bundle exec jekyll build --baseurl "${{ steps.pages.outputs.base_path }}" --trace
        env:
          JEKYLL_ENV: production
      - name: Upload artifact
        # Automatically uploads an artifact from the './_site' directory by default
        uses: actions/upload-pages-artifact@v1

  # Deployment job
  deploy:
    environment:
      name: github-pages
      url: ${{ steps.deployment.outputs.page_url }}
    runs-on: ubuntu-latest
    needs: build
    steps:
      - name: Deploy to GitHub Pages
        id: deployment
        uses: actions/deploy-pages@v1
